src/invidious/hashtag.cr

```diff
@@ -1,15 +1,42 @@
 module Invidious::Hashtag
   extend self
 
-  def fetch(hashtag : String, page : Int, region : String? = nil) : Array(SearchItem)
+  def fetch(hashtag : String, page : Int, region : String? = nil) : HashtagPage
     cursor = (page - 1) * 60
-    ctoken = generate_continuation(hashtag, cursor)
-
+    header = nil
     client_config = YoutubeAPI::ClientConfig.new(region: region)
-    response = YoutubeAPI.browse(continuation: ctoken, client_config: client_config)
+    # for any page besides the first page, get the list of videos
+    if cursor > 0
+      ctoken = generate_continuation(hashtag, cursor)
+      response = YoutubeAPI.browse(continuation: ctoken, client_config: client_config)
+    else
+      # get first page + header info
+      response = YoutubeAPI.browse("FEhashtag", params: get_hashtag_first_page(hashtag), client_config: client_config)
+      if header = response.dig?("header")
+        header = parse_item(header).as(HashtagHeader)
+      end
+    end
 
     items, _ = extract_items(response)
-    return items
+    return HashtagPage.new({
+      videos: items,
+      header: header,
+    })
+  end
+
+  def get_hashtag_first_page(hashtag : String)
+    object = {
+      "93:embedded" => {
+        "1:string" => hashtag,
+        "2:varint" => 0_i64,
+        "3:varint" => 1_i64,
+      },
+    }
+
+    return object.try { |i| Protodec::Any.cast_json(i) }
+      .try { |i| Protodec::Any.from_json(i) }
+      .try { |i| Base64.urlsafe_encode(i) }
+      .try { |i| URI.encode_www_form(i) }
```

Comment:
Could you merge that with the `generate_continuation` function below? The `93:embedded` object is the same between the two, so you can easily share the code.

Note: don't mind that `:2`, it's a leftover of protodec's parser. `93:2:embedded` and `93:embedded` are essentially the same thing.

Reply:
I have moved them to the same function but couldn't figure out how to reuse the object